### DISCUSSION TITLE
Low performance on Teensy 4.1, why?

### DISCUSSION THREAD

**Discussion by daywink** [moononournation/Arduino_GFX#251] [Q&A]
Hello, I have previously used a library called ILI9341_T4 on my Teensy 4.1 with an LVGL Benchmark demo and it gets average 1040 FPS, but if I use the same code and hardware, but using Arduino_GFX as my driver, I get 38 FPS average. Is this normal?

**Comment by moononournation** [comment]
Do you know the SPI speed they are running?

**Reply by daywink** [reply]
I tried 40mHz, 80mHz, 11mHz, and they all performed quite similarly on the ILI9341_T4 library but got some difference on the Arduino_GFX.

Edit: [Here is a video](https://www.youtube.com/watch?v=1s54A9MwCEQ) of the LVGL Music Demo running on ILI9341_T4 library, I didn't record Arduino_GFX, but it got about 30.

**Comment by moononournation** [comment]
ILI9341_T4 seems make many optimization for LVGL, I also recommend you stick to this library if you mainly using LVGL. The main goal of Arduino_GFX is portable and compatible across multiple Arduino IDE platform. You can run same code without rewrite when you change dev device, but it is not always fastest.
